sql/src/main/java/io/crate/analyze/DropTableAnalyzer.java

```diff
@@ -21,37 +21,39 @@
 
 package io.crate.analyze;
 
+import io.crate.exceptions.ResourceUnknownException;
 import io.crate.metadata.Schemas;
 import io.crate.metadata.TableIdent;
+import io.crate.metadata.doc.DocTableInfo;
 import io.crate.metadata.table.Operation;
-import io.crate.sql.tree.DefaultTraversalVisitor;
 import io.crate.sql.tree.DropTable;
-import io.crate.sql.tree.Node;
-import org.elasticsearch.common.inject.Inject;
-import org.elasticsearch.common.inject.Singleton;
 
-@Singleton
-public class DropTableStatementAnalyzer extends DefaultTraversalVisitor<DropTableAnalyzedStatement, Analysis> {
+import javax.annotation.Nullable;
+
+class DropTableAnalyzer {
 
     private final Schemas schemas;
 
-    @Inject
-    public DropTableStatementAnalyzer(Schemas schemas) {
+    DropTableAnalyzer(Schemas schemas) {
         this.schemas = schemas;
     }
 
-    @Override
-    public DropTableAnalyzedStatement visitDropTable(DropTable node, Analysis context) {
-        DropTableAnalyzedStatement statement = new DropTableAnalyzedStatement(schemas, node.ignoreNonExistentTable());
-        statement.table(TableIdent.of(node.table(), context.sessionContext().defaultSchema()));
-        if (!statement.noop()) {
-            Operation.blockedRaiseException(statement.tableInfo, Operation.DROP);
+    public DropTableAnalyzedStatement analyze(DropTable node, @Nullable String defaultSchema) {
+        TableIdent tableIdent = TableIdent.of(node.table(), defaultSchema);
+        DocTableInfo tableInfo = null;
+        boolean isNoop = false;
+        try {
+            tableInfo = schemas.getDropableTable(tableIdent);
+        } catch (ResourceUnknownException e) {
+            if (node.ignoreNonExistentTable()) {
+                isNoop = true;
+            } else {
+                throw e;
+            }
         }
-
-        return statement;
-    }
-
-    public AnalyzedStatement analyze(Node node, Analysis analysis) {
-        return process(node, analysis);
+        if (!isNoop) {
+            Operation.blockedRaiseException(tableInfo, Operation.DROP);
+        }
+        return new DropTableAnalyzedStatement(tableInfo, isNoop, node.ignoreNonExistentTable());
```

Comment:
ignoreNonExistentTable turns into a dropIfExists var which looks a bit odd. would be great to clean this up as well ;)